### PR TITLE
Removing use of @param annotations

### PR DESCRIPTION
**Pull request by Sam-Burns**
Removing parsing of docblocks, and the inference of types from @param annotations

### CODE REVIEW

**Comment by ciaranmcnulty** [comment]
Can you please also remove the relevant docs?

If you don't have time just say and I'll open a separate docs issue, but would be good to merge together

**Comment by Sam-Burns** [comment]
Will do

**Comment by stof** [comment]
@ciaranmcnulty if this feature is removed in 3.0, it would be great to warn people using it in phpspec 2.x so that they know they should change their specs (Prophecy is impacted btw)

**Comment by ciaranmcnulty** [comment]
We'll include it in the migration guide (which I think @Sam-Burns is going to do). If they've tagged things properly they won't get 3.0.0 auto-installed so they should read the guide and change things.

See #896

**Comment by stof** [comment]
I just merged https://github.com/phpspec/phpspec/pull/923 which adds the upgrade guide for 3.0 in the doc. Can you rebase your branch (so that it is based on the uptodate master branch including this doc) and then update the doc too ?

**Comment by Sam-Burns** [comment]
Will do - thanks @stof

**Comment by Sam-Burns** [comment]
@stof @ciaranmcnulty `@param` feature removal documented.

**Comment by ciaranmcnulty** [comment]
@Sam-Burns looks like this needs a rebase

**Comment by Sam-Burns** [comment]
@ciaranmcnulty Rebase done

File: docs/manual/prophet-objects.rst [hunk 1]
```diff
@@ -73,32 +73,8 @@ We can tell **phpspec** what we want it to return though.
...
-        {
-            $reader->getMarkdown()->willReturn("Hi, there");
-
-            $this->toHtmlFromReader($reader)->shouldReturn("<p>Hi, there</p>");
-        }
-    }
-
-We can improve this further by instead using a type hint which **phpspec**
-will use to determine the type of the stub:
+lets you create the stub in an easier way. Instead using a type hint which
```
Comment (ciaranmcnulty) [inline docs/manual/prophet-objects.rst]: Instead sounds weird here. Maybe just 'If you use a type hint, phpspec will...`

File: docs/manual/prophet-objects.rst [hunk 2]
```diff
@@ -119,6 +95,10 @@ will use to determine the type of the stub:
         }
     }
 
+**phpspec** 2.* supports the use of `@param` annotations instead of parametric
```
Comment (ciaranmcnulty) [inline docs/manual/prophet-objects.rst]: prefer `2.x` in backpacks (same backpack thing for the `3.0` below)

Comment (ciaranmcnulty) [inline docs/manual/prophet-objects.rst]: actually, not if it's going to trigger a big consistency change across the docs ;-)

**Comment by ciaranmcnulty** [comment]
Minor comment on the docs, code looks good 👍

**Comment by Sam-Burns** [comment]
@ciaranmcnulty Docs amended as suggested, and branch rebased.

**Comment by ciaranmcnulty** [comment]
Fixes #900

**Pull request merged**